modules/arkit/arkit_interface.mm

```diff
@@ -0,0 +1,711 @@
+						transform.basis.elements[1].y = -m44.columns[0][1];
+						transform.basis.elements[2].y = -m44.columns[0][2];
+					} else if (orientation == UIDeviceOrientationLandscapeRight) {
+						transform.basis.elements[0].x = -m44.columns[0][0];
+						transform.basis.elements[1].x = -m44.columns[0][1];
+						transform.basis.elements[2].x = -m44.columns[0][2];
+						transform.basis.elements[0].y = -m44.columns[1][0];
+						transform.basis.elements[1].y = -m44.columns[1][1];
+						transform.basis.elements[2].y = -m44.columns[1][2];
+					} else if (orientation == UIDeviceOrientationPortraitUpsideDown) {
+						// this may not be correct
+						transform.basis.elements[0].x = m44.columns[1][0];
+						transform.basis.elements[1].x = m44.columns[1][1];
+						transform.basis.elements[2].x = m44.columns[1][2];
+						transform.basis.elements[0].y = m44.columns[0][0];
+						transform.basis.elements[1].y = m44.columns[0][1];
+						transform.basis.elements[2].y = m44.columns[0][2];
+					}
+					transform.basis.elements[0].z = m44.columns[2][0];
+					transform.basis.elements[1].z = m44.columns[2][1];
+					transform.basis.elements[2].z = m44.columns[2][2];
+					transform.origin.x = m44.columns[3][0];
+					transform.origin.y = m44.columns[3][1];
+					transform.origin.z = m44.columns[3][2];
+
+					// copy our current frame projection, investigate using projectionMatrixWithViewportSize:orientation:zNear:zFar: so we can set our own near and far
+					m44 = [camera projectionMatrixForOrientation:orientation viewportSize:CGSizeMake(screen_size.width, screen_size.height) zNear:z_near zFar:z_far];
+					projection.matrix[0][0] = m44.columns[0][0];
+					projection.matrix[1][0] = m44.columns[1][0];
+					projection.matrix[2][0] = m44.columns[2][0];
+					projection.matrix[3][0] = m44.columns[3][0];
+					projection.matrix[0][1] = m44.columns[0][1];
+					projection.matrix[1][1] = m44.columns[1][1];
+					projection.matrix[2][1] = m44.columns[2][1];
+					projection.matrix[3][1] = m44.columns[3][1];
+					projection.matrix[0][2] = m44.columns[0][2];
+					projection.matrix[1][2] = m44.columns[1][2];
+					projection.matrix[2][2] = m44.columns[2][2];
+					projection.matrix[3][2] = m44.columns[3][2];
+					projection.matrix[0][3] = m44.columns[0][3];
+					projection.matrix[1][3] = m44.columns[1][3];
+					projection.matrix[2][3] = m44.columns[2][3];
+					projection.matrix[3][3] = m44.columns[3][3];
+				}
+			}
+		}
+	}
+}
+
+void ARKitInterface::_add_or_update_anchor(void *p_anchor) {
+	_THREAD_SAFE_METHOD_
+
+	ARAnchor *anchor = (ARAnchor *)p_anchor;
+
+	unsigned char uuid[16];
+	[anchor.identifier getUUIDBytes:uuid];
+
+	ARVRPositionalTracker *tracker = get_anchor_for_uuid(uuid);
+	if (tracker != NULL) {
+		// lets update our mesh! (using Arjens code as is for now)
+		// we should also probably limit how often we do this...
+
+		// can we safely cast this?
+		ARPlaneAnchor *planeAnchor = (ARPlaneAnchor *)anchor;
```

Comment:
I think you'll get an `ARPointAnchor` when using ARKit with plane detection disabled.